src/indicators/drought/wrsi.py

```diff
@@ -0,0 +1,309 @@
+            ):
+                if max_dekad == 33:
+                    max_dekad = 1
+                    max_year += 1
+                else:
+                    max_dekad += 1
+                download_wrsi(
+                    region=region,
+                    start_date=[max_year, max_dekad],
+                    end_date=[end_year, end_dekad],
+                    redownload=True,
+                )
+    else:
+        while not (
+            i_year > end_year or (i_year == end_year and i_dekad > i_dekad)
+        ):
+            _download_wrsi(
+                year=i_year, dekad=i_dekad, region=region, raw_dir=raw_dir
+            )
+            i_dekad += 1
+            if i_dekad > 33:
+                i_year += 1
+                i_dekad = 13
+
+
+_PROCESSED_FILENAME = "wrsi_{region}_{type}.nc"
+
+_PROCESSED_DIR = Path(
+    os.getenv("AA_DATA_DIR"), "public", "processed", "glb", "wrsi"
+)
+
+
+def _get_processed_path(region, type):
+    _PROCESSED_DIR.mkdir(parents=True, exist_ok=True)
+    return Path(
+        _PROCESSED_DIR,
+        _PROCESSED_FILENAME.format(region=region, type=type),
+    )
+
+
+# defined here: https://gis.stackexchange.com/questions/
+# 177447/defining-the-correct-aea-proj4string-for-
+# fewsnet-rainfall-data-southern-africa
+_WRSI_CRS = CRS.from_proj4(
+    "+proj=aea +lat_1=-19.0 +lat_2=21.0 +lat_0=1.0 "
+    "+lon_0=20 +x_0=0 +y_0=0 +ellps=clrk66 +units=m +no_defs"
+)
+
+
+def process_wrsi(region, type):
+    if region not in _VALID_REGION:
+        raise ValueError("`region` must be one of 'cropland' or 'rangeland'.")
+
+    raw_dir = _get_raw_dir(region)
+
+    types = {"current": "do", "extended": "eo", "anomaly": "er"}
+
+    if type in types.keys():
+        fp_type = types[type]
+    else:
+        raise ValueError(
+            "`type` must be one of 'current', 'extended', or 'anomaly'."
+        )
+
+    processed_path = _get_processed_path(region, type)
+
+    def _load_raw(fp):
+        arr = xr.open_rasterio(Path(raw_dir, fp))
+        dt = [_fp_date(fp)]
+
+        arr = arr.expand_dims(time=dt)
+        # round x dimension due to issues with
+        # concat and tiny numerical differences
+        arr = arr.assign_coords({"x": np.round(arr.x.values, 7)})
+        return arr
+
+    arrays = [
+        _load_raw(filename) for filename in raw_dir.glob(fp_type + "*.tif")
+    ]
+
+    arrays_merged = xr.concat(arrays, "time").sortby("time")
+
+    # change missing values from 0 to nan
+    arrays_merged = arrays_merged.where(arrays_merged.values > 0)
+    arrays_merged.attrs["_FillValue"] = np.NaN
+
+    # convert to dataset
+    ad = arrays_merged.to_dataset("band")
+    ad = ad.rename({1: "wrsi"})
+
+    # saving file
+    processed_path.unlink(missing_ok=True)
+
+    ad.to_netcdf(processed_path)
+
+    return ad
+
+
+def load_wrsi(region, type):
+    processed_path = _get_processed_path(region, type)
+    a = xr.open_dataset(processed_path).to_array("wrsi")
+    a = a.squeeze(drop=True)
```

Comment:
Yeah, I think it's nicer to explicitly save as dataset, because `DataArray.to_netcdf()` converts to a dataset on the backend, and wasn't sure if there was a better format to save in instead of `NetCDF` (prefer that to saving just a million band `.tif` for instance). But yes, will open with `rioxarray` and see how it's different! These comments are gold, thanks so much.